irods/session.py

```diff
@@ -350,8 +350,7 @@ def connection_timeout(self):
     @connection_timeout.setter
     def connection_timeout(self, seconds):
         self._cached_connection_timeout = seconds
-        if seconds is not None:
```

Comment:
Many systems also use 0 to indicate a non-value (or forever)… consider checking for 0 or None?

Reply:
That is a good point as evidently 0 means set nonblocking, and I  believe we want to disallow that.
None is, however, in my opinion should be permissible  as it is (for all intents) equivalent to a very large number.

Reply:
In other words `session.connection_timeout = None` is clear and correct in intent, if you take it to mean eliminate timeouts totally for the given session.

Reply:
Agreed that `None` conveys the intent.

I was suggesting that if this is a user-facing setting, that setting it to either `0` or `None` would be equivalent and correct.  If this is not user-facing, then... carry on.

Reply:
The docs encourage the use of:

```
session_object.connection_timeout = <some_number> 
```

as a way to dictate how long the client socket should wait until declaring a server peer as down and giving up, as it were.

So, yes, it is user-facing...

But `0` and `None` are not equivalent here, actually. `0` would damage the socket for further use by the PRC, by turning ono non-blocking behavior (ie null values could now be read from the socket when the it has not yet received any bytes of the expected server response) whereas `None` declares we never want to declare the server peer down.  i.e. "turn off timeouts completely"

Reply:
If Windows is a different case, though, I should evaluate

Reply:
No worries - if `None` is the only correct answer - please consider leaving a comment somewhere user-facing... that will let them know that `0` behaves differently and why.

Reply:
Sounds good. I'll update the docs to caution against setting 0 (which doesn't yet cause an error , but should) or even too low a value, as the latter would have an effect down the line of raising a NetworkError the first time the server "responds too slowly", which (as long as the server is alive and still working) would be technically incorrect practice on the application writer's part.